python_modules/dagster/dagster/_scheduler/scheduler.py

```diff
@@ -648,8 +648,8 @@ def launch_scheduled_runs_for_schedule_iterator(
                 # Plan to run the same tick again right away
                 yield ScheduleIterationTimes(
                     cron_schedule=external_schedule.cron_schedule,
-                    next_iteration_timestamp=now_timestamp,
-                    last_iteration_timestamp=now_timestamp,
+                    next_iteration_timestamp=schedule_time.timestamp(),
+                    last_iteration_timestamp=schedule_time.timestamp(),
```

Comment:
oof this is subtle, good catch

I think its worth throwing a comment in here explaining this situation, just to build awareness of this flavor of problem for future people working on this